wgpu-hal/src/gles/adapter.rs

```diff
@@ -293,13 +339,39 @@ impl super::Adapter {
             ver >= (3, 1),
         );
 
+        let mut workarounds = super::Workarounds::empty();
+        let r = renderer.to_lowercase();
+        // Check for Mesa sRGB clear bug. See
+        // [`super::PrivateCapabilities::MESA_I915_SRGB_SHADER_CLEAR`].
+        if r.contains("mesa")
+            && r.contains("intel")
+            && r.split(&[' ', '(', ')'][..])
+                .any(|substr| substr.len() == 3 && substr.chars().nth(2) == Some('l'))
+        {
+            log::warn!(
+                "Detected skylake derivative running on mesa i915. Clears to srgb textures will \
+                use manual shader clears."
+            );
+            workarounds.set(super::Workarounds::MESA_I915_SRGB_SHADER_CLEAR, true);
+        }
+
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            private_caps.set(super::PrivateCapabilities::INDEX_BUFFER_ROLE_CHANGE, true);
+            private_caps.set(super::PrivateCapabilities::BUFFER_STORAGE, true);
```

Comment:
I guess we need move `gfx-hal` buffer storage support check here:
https://github.com/gfx-rs/gfx/blob/3ee1ca9ba486b166a52765024d8d149cbb28d486/src/backend/gl/src/info.rs#L765-L769

Reply:
Updated. :+1: